kernel/kernel-api/src/main/java/io/delta/kernel/engine/CommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,152 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.engine;
+
+import java.util.Map;
+
+import io.delta.kernel.annotation.Evolving;
+import io.delta.kernel.commit.Commit;
+import io.delta.kernel.commit.CommitResponse;
+import io.delta.kernel.commit.GetCommitsResponse;
+import io.delta.kernel.commit.UpdatedActions;
+import io.delta.kernel.commit.actions.AbstractMetadata;
+import io.delta.kernel.commit.actions.AbstractProtocol;
+import io.delta.kernel.data.Row;
+import io.delta.kernel.utils.CloseableIterator;
+
+/**
+ * Provides coordinated commits related functionalities to Delta Kernel.
+ *
+ * @since 3.0.0
+ */
+@Evolving
+public interface CommitCoordinatorClientHandler {
+
+    /**
+     * API to register the table represented by the given `logPath` at the provided
+     * currentTableVersion with the commit coordinator this commit coordinator client represents.
+     * <p>
+     * This API is called when the table is being converted from a file system table to a
+     * coordinated-commit table.
+     * <p>
+     * When a new coordinated-commit table is being created, the currentTableVersion will be -1 and
+     * the upgrade commit needs to be a file system commit which will write the backfilled file
+     * directly.
+     *
+     * @param logPath         The path to the delta log of the table that should be converted
+     * @param currentVersion  The currentTableVersion is the version of the table just before
+     *                        conversion. currentTableVersion + 1 represents the commit that
+     *                        will do the conversion. This must be backfilled atomically.
+     *                        currentTableVersion + 2 represents the first commit after conversion.
+     *                        This will go through the CommitCoordinatorClient and the client is
+     *                        free to choose when it wants to backfill this commit.
+     * @param currentMetadata The metadata of the table at currentTableVersion
+     * @param currentProtocol The protocol of the table at currentTableVersion
+     * @return A map of key-value pairs which is issued by the commit coordinator to identify the
+     * table. This should be stored in the table's metadata. This information needs to be
+     * passed to the {@link #commit}, {@link #getCommits}, and {@link #backfillToVersion}
+     * APIs to identify the table.
+     */
+    Map<String, String> registerTable(
```

Comment:
If we are not going to use this API, can we add this in followup PRs?

Reply:
I think it's the interface API of `CommitCoordinatorClient` defined in the Storage module?